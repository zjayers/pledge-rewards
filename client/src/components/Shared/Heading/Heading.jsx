import React from 'react';
import Heading from './Heading.style';

export default ({ heavy, children }) => {
  return <Heading heavy={heavy}>{children}</Heading>;
};
