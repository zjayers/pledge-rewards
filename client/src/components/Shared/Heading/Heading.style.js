import styled from 'styled-components';

export default styled.h2`
  font-size: ${(props) => props.theme.fontSizes.medium};
  color: ${(props) => props.theme.colors.text};
  font-weight: ${(props) => {
    if (props.heavy) {
      return 700;
    }
    return 400;
  }};
  line-height: 1.4;
  margin-bottom: 5px;
`;
