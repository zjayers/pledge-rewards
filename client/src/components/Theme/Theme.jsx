import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
