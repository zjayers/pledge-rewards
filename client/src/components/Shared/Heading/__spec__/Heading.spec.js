import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Heading from '../Heading';
import Theme from '../../../Theme/Theme';

describe('Heading Component', () => {
  let component;

  beforeEach(() => {
    component = mount(
      <Theme>
        <Heading />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
