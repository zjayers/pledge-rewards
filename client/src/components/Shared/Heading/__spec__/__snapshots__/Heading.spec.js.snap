// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default>
      <Headingstyle___default>
        <h2
          className="Headingstyle___default-dlpxpq-0 iErRtB"
        />
      </Headingstyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;
